spec/requests/users_controller_spec.rb

```diff
@@ -5186,6 +5186,90 @@ def revoke(user, skip_remote: false)
     end
   end
 
+  describe "#reset_recent_searches" do
+    fab!(:user) { Fabricate(:user) }
+
+    it 'does nothing for anon' do
+      delete "/u/recent-searches.json"
+      expect(response.status).to eq(403)
+    end
+
+    it 'works for logged in user' do
+      sign_in(user)
+      delete "/u/recent-searches.json"
+
+      expect(response.status).to eq(200)
+      user.reload
+      expect(user.user_option.oldest_search_log_date).to be_within(5.seconds).of(1.second.ago)
```

Comment:
Hmm I feel this has the potential to be flaky or inaccurate. Can we perhaps set an initial value for `oldest_search_log_date` and check that the value changed after the `delete` request?

Reply:
Sure, I updated the test to set `oldest_search_log_date` to `nil` first and then to a set date. It's useful to check both scenarios, a user that never clears their log will always have this column set to `nil`.